test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1

```diff
@@ -116,6 +116,16 @@ Describe 'Positive Parse Properties Tests' -Tags "CI" {
         class C9b { [System.Collections.Generic.List[C9b]] f() { return [C9b]::new() } }
         $c9b = [C9b]::new().f()
         It "Expected a System.Collections.Generic.List[C9b] returned" {  $c9b -is [System.Collections.Generic.List[C9b]] | Should -BeTrue }
+        It 'Methods returning object should return $null if no output was produced' {
+            class Foo {
+                [object] Bar1() { return & {} }
+                static [object] Bar2() { return & {} }
+            }
+            # Test instance method
+            [Foo]::new().Bar1() | Should -BeNullOrEmpty
```

Comment:
We could explicitly check result type. I believe it make sense to use `testcase` for two return types (object and int/string) to ensure that default(T) works as expected.

Reply:
The *only* time this code path is hit is if the method returns `[object]`. All of the other cases are already handled elsewhere.  And the object returned is null. How do you propose checking it's type?

Reply:
Sorry, it was my misunderstanding of default(T).